src/checker.rs

```diff
@@ -49,7 +50,50 @@ mod tests {
         let profile = "release";
 
         let expected = format!("{}/target/{}/deps", pwd, profile);
-        assert_eq!(expected, get_base_path().unwrap());
+        let exe = env::current_exe();
+
+        match exe {
+            Ok(path) => assert_eq!(expected, get_base_path(path).unwrap()),
+            Err(_) => assert_eq!(true, false),
+        };
+    }
+
+    #[test]
+    fn test_download_and_unpack() -> Result<()> {
+        let base_url = generate_base_url("2.0.3");
+        let exe = env::current_exe();
+        let mut base_path = String::from("");
+        match exe {
+            Ok(path) => base_path = get_base_path(path).unwrap(),
+            Err(_) => assert_eq!(true, false),
+        }
+
+        let filename = generate_filename(
+            &get_os_type(sys_info::os_type()?).unwrap(),
+            get_architecture().unwrap(),
+        );
+
+        let result = download(&base_url, &base_path, &filename);
+        let tar_path = format!("{}/{}.tar.gz", base_path, filename);
+        assert_eq!((), result.unwrap());
+        assert_eq!(true, path_exists(&tar_path));
+
+        let result = unpack(&tar_path, &base_path);
```

Comment:
`unpack` can be tested in isolation by creating some file, create a tar.gz archive containing the file and assert that unpacking the archive creates the initial file and nothing else.

Reply:
Would this be better? https://github.com/editorconfig-checker/editorconfig-checker.rust/pull/7/commits/5f1846f8f32192b63abac199b22cf70be0791b5f

Or did I misunderstood something?

Reply:
It has it's own test now \o/